### PR TITLE
Cb system configs

### PR DESCRIPTION
**Pull request by jamhed**

### CODE REVIEW

File: applications/crossbar/priv/api/swagger.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
             "type": "object"
         },
         "account_config.cdr": {
+            "additionalProperties": false,
```
Comment (fenollp) [inline applications/crossbar/priv/api/swagger.json]: Hey so this is weird https://github.com/json-schema-org/json-schema-spec/blob/f7bb080b0c6af0b23505cdcc0143bccf001b6d40/jsonschema-validation.xml#L509

**Comment by jamhed** [comment]
@fenollp, what's about https://spacetelescope.github.io/understanding-json-schema/reference/object.html?

**Comment by fenollp** [comment]
Agreed. Time to write some documentation! Lucky you!

**Comment by jamhed** [comment]
This is actually based on https://github.com/2600hz/kazoo/pull/3224, therefore all these changes. Once #3224 get merged I'll rebase.

File: applications/crossbar/doc/ref/system_configs.md [hunk 2]
```diff
@@ -46,6 +46,16 @@ curl -v -X POST \
     http://{SERVER}:8000/v2/system_configs/{SYSTEM_CONFIG_ID}
 ```
 
+#### Patch
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/ref/system_configs.md]: Please migrate these changes to `crossbar/doc/system_config.md` and flesh them out appropriately

File: core/kazoo_json/src/kz_json.erl [hunk 3]
```diff
@@ -1218,6 +1218,22 @@ diff(J1, J2) ->
     ?JSON_WRAPPER(L2) = flatten(J2),
     expand(from_list(L1 -- L2)).
 
+-spec flatten_schema(object()) -> object().
+flatten_schema(?JSON_WRAPPER([]) = Empty) -> Empty;
+flatten_schema(?JSON_WRAPPER(L) = Schema) when is_list(L) ->
+    from_list(lists:flatten(flatten_props(get_value(<<"properties">>, Schema), [], Schema))).
```
Comment (jamesaimonetti) [inline core/kazoo_json/src/kz_json.erl]: This should be in kz_json_schema - that module understands JSON schema documents. kz_json does not.

**Pull request closed without merging**